frontend/packages/dev-console/src/components/pipelines/pipeline-builder/hooks.ts

```diff
@@ -135,7 +135,10 @@ export const useNodes = (
 
   // TODO: Fix id collisions then remove this utility; we shouldn't need to trim the tasks
   const noDuplicates = (resource: PipelineResourceTask) =>
-    !taskGroupRef.current.tasks.find((pt) => pt.name === resource.metadata.name);
+    !taskGroupRef.current.tasks.find(
+      (pt) => pt.taskRef.name === resource.metadata.name && pt.taskRef.kind === resource.kind,
```

Comment:
We can't do `resource.metadata.name` against `pt.taskRef.name` because this will prevent any double uses of a given task. This isn't the end of the world, but likely will be a problem for some use-cases.

However! With my recent changes suggested, I think we can actually get rid of this function since we'll always will generate a new name.